sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4294,6 +4294,31 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         Row(3, 2, 6) :: Nil)
     }
   }
+
+  test("SPARK-38118: Func(wrong_type) in the HAVING clause should throw data mismatch error") {
```

Comment:
Thank you for adding this. Do you think you can add a test coverage in `catalyst` module too because your main contribution is inside `catalyst` module.

We recommend to keep the patch main code and test code in **the same module closely** in order to maintain the modules more completely by themselves and to test them more easily.

Reply:
ideally we should add UT in catalyst as well, such as `AnalysisSuite`. Not sure how hard it is for this case though...

Reply:
I found that I can easily put the existing tests into `AnalysisSuite`. Per my understanding you are suggesting we put test into `catalyst` as this is change in `catalyst`.  So I ended up with **moving** test cases to `AnalysisSuite`. Please let me know if you think we should also have tests in `SQLQuerySuite`.